docs/en/creating-widgets/supplemental.md

```diff
@@ -0,0 +1,972 @@
+
+For complete applications, simple widgets also allows for easier understanding of all constituent components, as well as how they get combined.
+
+Together these benefits lead to simpler ongoing maintenance and an ultimate reduction in the overall cost of building and running an application.
+
+## Basic widget structure
+
+At their heart, widgets are simply render functions which return VDOM nodes that form the widget's structural representation within a web page. However, applications typically require more logic than a simple list of HTML elements, so meaningful widgets are usually comprised of more than just a simple render function.
+
+Widgets are typically housed within their own self-named TypeScript modules, with the widget definition as the default export from each module.
+
+The simplest way of representing widgets is based on plain functions, starting from a render function factory definition. Dojo provides a `create()` primitive in the `@dojo/framework/core/vdom` module that allows authors to define their widget render function factories. Named render functions are preferred as they can help with debugging, but this is not a requirement; widgets can also by identified via an exported variable holding their factory definition.
+
+Dojo optionally supports class-based widgets for applications that prefer the structure of classes over functions. Such widgets inherit from `WidgetBase`, provided by the `@dojo/framework/core/WidgetBase` module, and are required to implement a `render()` method.
+
+The following example shows a trivial yet complete widget within a Dojo application:
+
+> src/widgets/MyWidget.ts
+
+**Function-based Dojo widget variant:**
+
+```ts
+import { create } from '@dojo/framework/core/vdom';
+
+const factory = create();
+
+export default factory(function MyWidget() {
+	return [];
+});
+```
+
+**Class-based Dojo widget variant:**
+
+```ts
+import WidgetBase from '@dojo/framework/core/WidgetBase';
+
+export default class MyWidget extends WidgetBase {
+	protected render() {
+		return [];
+	}
+}
+```
+
+Because this widget returns an empty array from its render function, it has no structural representation within an application's output. Widgets typically [return one or more virtual DOM nodes](#virtual-nodes-example) in order to provide meaningful structure within the application's HTML output.
+
+The process of translating virtual DOM nodes to output on a web page is handled by [Dojo's rendering system](#rendering-in-dojo).
+
+## Widget styling
+
+Styling of a widget's DOM output is handled via CSS, with relevant style classes stored in a CSS module file parallel to the widget's TypeScript module. Styling is identical for both function- and class-based widget variants. This topic is described in detail within the [Styling and Theming reference guide](../styling-and-theming/supplemental.md).
+
+## Scaffolding widgets
```

Comment:
I'm beginning to think we don't need this section in the guide... what do others think?

Reply:
Removed in favor of an upcoming Getting Started guide